main.go

```diff
@@ -549,7 +557,7 @@ func (c *controller) sync(ctx context.Context) {
 		return
 	}
 
-	statefulsets := make(map[string]*appsv1.StatefulSet)
```

Comment:
The original assumption is one hashring string has only one statefulset in this function. This PR breaks that assumption. What's the reason?

Reply:
in our setup, one hashring config includes multiple statefulset.
Originally one hashring config only includes one statefulset, and I see they set it up each statefulset has their own hashring config. 
@jnyi do you know why we set up in a single hashring config instead of one for each sts?

Reply:
it should be 1 hashrings, the key of multiple hashrings are tenant, see https://thanos.io/tip/components/receive.md/#az-aware-ketama-hashring-experimental